drivers/input/input_esp32_touch_sensor.c

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT espressif_touch_sensor
+
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <esp_err.h>
+#include <soc/soc_pins.h>
+#include <soc/periph_defs.h>
+#include <hal/touch_sensor_types.h>
+#include <hal/touch_sensor_hal.h>
+#include <driver/rtc_io.h>
+#include <esp_intr_alloc.h>
+
+LOG_MODULE_REGISTER(espressif_touch_sensor, CONFIG_INPUT_LOG_LEVEL);
+
+#if defined(CONFIG_COUNTER_RTC_ESP32)
+#     error "Conflict detected: COUNTER_RTC_ESP32 enabled"
+#endif /* defined(CONFIG_COUNTER_RTC_ESP32) */
+
+#define ESP32_SCAN_DONE_MAX_COUNT 5
+
+#if defined(CONFIG_SOC_SERIES_ESP32)
+#	define ESP32_RTC_INTR_MSK RTC_CNTL_TOUCH_INT_ST_M
+#elif defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+#	define ESP32_RTC_INTR_MSK 			\
+		RTC_CNTL_TOUCH_DONE_INT_ST_M | 		\
+		RTC_CNTL_TOUCH_ACTIVE_INT_ST_M |	\
+		RTC_CNTL_TOUCH_INACTIVE_INT_ST_M |	\
+		RTC_CNTL_TOUCH_SCAN_DONE_INT_ST_M |	\
+		RTC_CNTL_TOUCH_TIMEOUT_INT_ST_M
```

Comment:
Can you parenthesize these? Prob not necessary but just to be safe, good practice anyway